src/nvim/os/pty_process_unix.c

```diff
@@ -171,7 +171,13 @@ int pty_process_spawn(PtyProcess *ptyproc)
   Process *proc = (Process *)ptyproc;
   assert(proc->err.closed);
   uv_signal_start(&proc->loop->children_watcher, chld_handler, SIGCHLD);
-  ptyproc->winsize = (struct winsize){ ptyproc->height, ptyproc->width, 0, 0 };
+  ioctl(1, TIOCGWINSZ, &ptyproc->winsize);
+  ptyproc->winsize = (struct winsize){
+    ptyproc->height,
+    ptyproc->width,
+    ptyproc->winsize.ws_xpixel / ptyproc->winsize.ws_col * ptyproc->width,
+    ptyproc->winsize.ws_ypixel / ptyproc->winsize.ws_row * ptyproc->height,
```

Comment:
use parens to make the order of operations obvious
```suggestion
    (ptyproc->winsize.ws_xpixel / ptyproc->winsize.ws_col) * ptyproc->width,
    (ptyproc->winsize.ws_ypixel / ptyproc->winsize.ws_row) * ptyproc->height,
```